### PR TITLE
Add new hierarchy analyzer to DebugUtils

### PR DESCRIPTION
**Pull request by Jannify**
[Example](https://cdn.discordapp.com/attachments/325084661011316737/370613242818461711/unknown.png)

### CODE REVIEW

File: NitroxClient/Unity/Helper/DebugUtils.cs [hunk 1]
```diff
@@ -7,6 +7,46 @@ namespace NitroxClient.Unity.Helper
...
+            Log.Debug("------------------------------------");
+            StartHierachy(obj);
+            Log.Debug("------------------------------------");
+        }
+
+        private static void StartHierachy(GameObject obj)
+        {
+            prefix = prefix + "    ";
+            Log.Debug(GetInfoOfGamobject(obj));
+            for (int i = 0; i < obj.transform.childCount; i++)
```
Comment (MarijnS95) [inline NitroxClient/Unity/Helper/DebugUtils.cs]: Why the indices? The other `DumpGameObject` example clearly shows that you can do this in tidy `foreach` loop. If `Transform` were using a linked list, `GetChild(i)` would be immensely slow.

File: NitroxClient/Unity/Helper/DebugUtils.cs [hunk 2]
```diff
@@ -7,6 +7,46 @@ namespace NitroxClient.Unity.Helper
...
+
+        public static void DumpHierarchy(GameObject obj)
+        {
+            Log.Debug("Analyze gameobject: ");
+            Log.Debug("------------------------------------");
+            StartHierachy(obj);
+            Log.Debug("------------------------------------");
+        }
+
+        private static void StartHierachy(GameObject obj)
```
Comment (MarijnS95) [inline NitroxClient/Unity/Helper/DebugUtils.cs]: Are you using this typo intentionally to distinguish `StartHierarchy` from `StartHierachy`?

File: NitroxClient/Unity/Helper/DebugUtils.cs [hunk 3]
```diff
@@ -7,6 +7,46 @@ namespace NitroxClient.Unity.Helper
...
+        private static string GetComponentsOfGamobject(GameObject obj)
+        {
+            string list = "";
+            foreach (Component item in obj.GetComponents<Component>())
+            {
+                list += item.GetType() + " | ";
+            }
+            if (list.Length >= 3)
+            {
+                list = list.Substring(0, list.Length-3);
```
Comment (MarijnS95) [inline NitroxClient/Unity/Helper/DebugUtils.cs]: Yuck. Use something like `string.Join` to add the separator.

This entire function can be converted into a oneliner:
```C#
return string.Join(" | ", obj.GetComponents<Component>().Select(item => item.GetType().ToString()).ToArray());
```

**Comment by Sunrunner37** [comment]
I think we will continue using the existing gameObject debug dumper.  Feel free to keep this locally for your own use.

**Pull request closed without merging**